### PR TITLE
User-Agent Client Hints blog

### LINKED ISSUE #3096: content: User-Agent Client Hints intro [2020-06-04]
* [draft](https://docs.google.com/document/d/1vcgh_SzydOXmzGBGlLhOHpW73tpG-gNeRUymFuGvpf8/)
* [thread](https://groups.google.com/a/google.com/g/webdev-content/c/LCCz35AOYMw)

### PR DESCRIPTION
**Pull request by rowan-m**
Fixes #3096 
Reviewed draft: https://docs.google.com/document/d/1vcgh_SzydOXmzGBGlLhOHpW73tpG-gNeRUymFuGvpf8/edit

### CODE REVIEW

**Comment by rowan-m** [comment]
FYI @yoavweiss

**Comment by rowan-m** [comment]
I'll merge once everyone's happy. Just want to coordinate on publishing.

**Comment by kaycebasques** [comment]
Reviewing now…

File: src/site/content/en/blog/user-agent-client-hints/index.md [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+
+```text
+Viewport-Width: 460
+Width: 230
+```
+
+The server can choose to vary its responses, for example by serving images at an
+appropriate resolution.
+
+{% Aside 'objective' %}
```
Comment (kaycebasques) [inline src/site/content/en/blog/user-agent-client-hints/index.md:131]:
```suggestion
{% Aside %}
```
Using the "objective" styling is a bit of a stretch here and distracted me. We usually use that to indicate the main objective of a tutorial. Removing `'objective'` will style this as a [note aside](https://web.dev/handbook/web-dev-components/#note-asides)

image

File: src/site/content/en/blog/user-agent-client-hints/index.md [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+  There are ongoing discussions on enabling Client Hints on an initial request,
+  but you should be considering responsive design or progressive enhancement
```
Comment (kaycebasques) [inline src/site/content/en/blog/user-agent-client-hints/index.md:133]:
```suggestion
  but you should be considering [responsive design](/responsive-web-design-basics) or progressive enhancement
```

File: src/site/content/en/blog/user-agent-client-hints/index.md [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+  },
+  {
+    brand: 'Google Chrome',
+    version: '84',
+  },
+];
+
+console.log(navigator.userAgentData.mobile);
+
+false;
```
Comment (kaycebasques) [inline src/site/content/en/blog/user-agent-client-hints/index.md:255]:
```suggestion
console.log(navigator.userAgentData.brands);

// [
//   {
//     brand: 'Chromium',
//     version: '84',
//   },
//   {
//     brand: 'Google Chrome',
//     version: '84',
//   },
// ];

console.log(navigator.userAgentData.mobile);

// false;
```
I'm not married to this exact convention but I think we need to provide some indicator that this is output. Because on my first readthrough when I saw `false` I thought you had accidentally copied some unrelated code.

File: src/site/content/en/blog/user-agent-client-hints/index.md [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+     "uaFullVersion"])
+  .then(ua => { console.log(ua) });
+
+{
+  "architecture": "x86",
+  "model": "",
+  "platform": "Linux",
+  "platformVersion": "",
+  "uaFullVersion": "84.0.4143.2"
+}
```
Comment (kaycebasques) [inline src/site/content/en/blog/user-agent-client-hints/index.md:278]:
```suggestion
// {
//   "architecture": "x86",
//   "model": "",
//   "platform": "Linux",
//   "platformVersion": "",
//   "uaFullVersion": "84.0.4143.2"
// }
```

File: src/site/content/en/blog/user-agent-client-hints/index.md [hunk 5]
```diff
@@ -0,0 +1,434 @@
...
+```
+
+### Demo
+
+You can try out both the headers and the JavaScript API on your own device at
+[user-agent-client-hints.glitch.me](https://user-agent-client-hints.glitch.me).
+
+<div class="glitch-embed-wrap" style="height: 420px; width: 100%;">
```
Comment (kaycebasques) [inline src/site/content/en/blog/user-agent-client-hints/index.md:286]: Feel free to adjust the `height` if you want to show more of the demo without needing to scroll

File: src/site/content/en/blog/user-agent-client-hints/index.md [hunk 6]
```diff
@@ -0,0 +1,434 @@
...
+    src="https://glitch.com/embed/#!/embed/user-agent-client-hints?path=README.md&previewSize=100"
+    title="user-agent-client-hints on Glitch"
+    allow="geolocation; microphone; camera; midi; vr; encrypted-media"
+    style="height: 100%; width: 100%; border: 0;">
+  </iframe>
+</div>
+
+{% Aside %}
+  The demo assumes that you're running a browser based on Chromium 84 or above.
+{% endAside %}
```
Comment (kaycebasques) [inline src/site/content/en/blog/user-agent-client-hints/index.md:297]: It would be nice to also say this within the demo UI

File: src/site/content/en/blog/user-agent-client-hints/index.md (hunk 5 above)
Comment (rowan-m) [reply src/site/content/en/blog/user-agent-client-hints/index.md:286]: Good call! Adjusting to `600px`.

File: src/site/content/en/blog/user-agent-client-hints/index.md (hunk 6 above)
Comment (rowan-m) [reply src/site/content/en/blog/user-agent-client-hints/index.md:297]: Will do. I need to update the demo readme and metadata too.

File: src/site/content/en/blog/user-agent-client-hints/index.md (hunk 3 above)
Comment (rowan-m) [reply src/site/content/en/blog/user-agent-client-hints/index.md:255]: I'm trying a code comment before the input and output lines to make a clearer separation and still get the syntax highlighting on the output.

**Comment by rowan-m** [comment]
A few last minute changes to drop in, then will pass for a final check.

**Comment by kaycebasques** [comment]
@rowan-m please merge this yourself before EOD Monday 6/29 (PST)

**Comment by rowan-m** [comment]
Will do!

File: src/site/content/en/blog/user-agent-client-hints/index.md [hunk 7]
```diff
@@ -149,13 +149,13 @@ all works.
...
-  [compatibility
-  concerns](https://bugs.chromium.org/p/chromium/issues/detail?id=1091285)
-  are resolved.
-{% endAside %}
+User-Agent Client Hints will only be enabled gradually on Chrome Stable as
+[compatibility
+concerns](https://bugs.chromium.org/p/chromium/issues/detail?id=1091285) are
+resolved. To force the functionality on for testing, ensure you:
+
+  * are on Chome 84 **beta** or equivalent
```
Comment (yoavweiss) [inline src/site/content/en/blog/user-agent-client-hints/index.md:157]: s/Chome/Chrome/
